kyaml/kio/byteio_writer.go

```diff
@@ -125,7 +128,29 @@ func (w ByteWriter) Write(nodes []*yaml.RNode) error {
 	for i := range nodes {
 		items.Content = append(items.Content, nodes[i].YNode())
 	}
-	err := errors.Wrap(encoder.Encode(doc))
+	rNode := yaml.RNode{}
+	rNode.SetYNode(doc)
+	err := w.encode(encoder, &rNode)
 	yaml.UndoSerializationHacksOnNodes(nodes)
 	return err
 }
+
+// encode encodes the input RNode to appropriate node format based on file path extension
+func (w ByteWriter) encode(encoder *yaml.Encoder, node *yaml.RNode) error {
+	for _, g := range JSONMatch {
+		if match, err := filepath.Match(g, filepath.Ext(w.Path)); err != nil {
```

Comment:
This seems redundant.  Either `filepath.Match(g, w.Path)` or `filepath.Ext(g) == filepath.Ext(w.Path)` seem simpler.

Reply:
Done.